WordPress/Classes/Models/Notifications/NotificationSettings.swift

```diff
@@ -0,0 +1,199 @@
+import Foundation
+
+
+/**
+*  @class           NotificationSettings
+*  @brief           The goal of this class is to encapsulate all of the User's Notification Settings in a generic way.
+*                   Settings are grouped into different Channels. A Channel is considered anything that might 
+*                   produce Notifications: a WordPress blog, Third Party Sites or WordPress.com.
+*                   Each channel may support different streams, such as: Email + Push Notifications + Timeline.
+*/
+
+public class NotificationSettings
+{
+    /**
+    *  @details Represents the Channel to which the current settings are associated.
+    */
+    public let channel  : Channel
+    
+    /**
+    *  @details Contains an array of the available Notification Streams.
+    */
+    public let streams  : [Stream]
+    
+    /**
+    *  @details Maps to the associated blog, if any.
+    */
+    public let blog     : Blog?
+    
+    
+    
+    /**
+    *  @details     Designated Initializer
+    *  @param       channel     The related Notifications Channel
+    *  @param       streams     An array of all of the involved streams
+    *  @param       blog        The associated blog, if any
+    */
+    public init(channel: Channel, streams: [Stream], blog: Blog?) {
+        self.channel = channel
+        self.streams = streams
+        self.blog    = blog
```

Comment:
@koke this feature is split among several Pull Requests. The first one being Service Layer (this PR).

The Interface Layer was already reviewed + merged back into the master [NCC's branch, which is available here](https://github.com/wordpress-mobile/WordPress-iOS/tree/feature/3807-notifications-control-center).

Thanks for checking!